analyzers/src/SonarAnalyzer.Common/Helpers/SyntaxNodeExtensions.cs

```diff
@@ -37,6 +37,22 @@ public static bool ToStringContainsEitherOr(this SyntaxNode node, string a, stri
         return toString.Contains(a) || toString.Contains(b);
     }
 
+    public static string GetMappedFilePath(this SyntaxNode root)
+    {
+        if (root.DescendantTrivia().FirstOrDefault() is { RawKind: (ushort)SyntaxKindEx.PragmaChecksumDirectiveTrivia } pragmaChecksum)
```

Comment:
Educational: Is there any reason why you are casting the syntaxkind to `ushort` instead of `int`? While I understand that it might save some memory, I wonder if it might have some impact on performance (most probably negligible). I read somewhere that modern CPUs are optimized for working with 32-bit integers but it might be very likely that I made that up :smile:

Reply:
The main reason is that the base type of SyntaxKind enum is already `ushort`. See https://github.com/dotnet/roslyn/blob/main/src/Compilers/CSharp/Portable/Syntax/SyntaxKind.cs#L9